.github/actions/ckan-setup/action.yml

```diff
@@ -0,0 +1,88 @@
+name: "Setup CKAN"
+
+inputs:
+  ckan-version:
+    description: CKAN version to install. set to 'latest' to always install the most recent version
+    default: 'latest'
+
+  ckan-url:
+    description: URL of the CKAN github repository to pull from. Defaults to the mainline version
+    default: 'https://github.com/KSP-CKAN/CKAN'
+
+  ckan-compatible-versions:
+    description: KSP versions to mark as compatible
+    type: string
+    default: |
+      1.12
+      1.11
+      1.10
+      1.9
+      1.8
+
+  ckan-filters:
+    description: CKAN filters to enable
+    type: string
+    default: |
+      .dds
+      .png
+      .bmp
+      .mu 
+      .mbm
+      .jpg
+      .wav
+
+  dependency-identifiers:
+    description: CKAN mod identifiers to install
+    type: string
+
+runs:
+  using: composite
+  steps:
+    - name: Install CKAN
+      shell: bash
+      run: |
+        set -u
+        mkdir -p "$CKAN_DIR"
+        echo "$CKAN_DIR" >> "$GITHUB_PATH"
+        curl -L https://github.com/KSP-CKAN/CKAN/releases/${{ env.CKAN_VERSION_SLUG }}/ckan.exe -o "$CKAN_DIR/ckan.exe"
+        sudo echo "mono $CKAN_DIR/ckan.exe \$@" > "$CKAN_DIR/ckan"
+        sudo chmod +x "$CKAN_DIR/ckan"
+        ${{ runner.debug && 'cat $CKAN_DIR/ckan' }}
+      env:
+        CKAN_VERSION_SLUG: ${{ format(inputs.ckan-version == 'latest' && 'latest/download' || 'download/v{0}', inputs.ckan-version)}}
+        CKAN_DIR: '${{ github.workspace }}/ckan'
+
+    - name: Setup CKAN Instance
+      shell: bash
+      run: |
+        ${{ runner.debug && 'echo "$PATH"' }}
+        ${{ runner.debug && 'ls -la $GITHUB_WORKSPACE/ckan' }}
+        ckan instance fake --set-default KSP ${{ env.KSPRoot }} 1.12.5 --game KSP --MakingHistory 1.9.1 --BreakingGround 1.7.1
+        ckan update
+
+    - name: Setup CKAN Compatible Versions
+      shell: bash
+      if: inputs.ckan-compatible-versions != ''
+      run: |
+        for version in $VERSIONS; do
+        ckan compat add "$version"
+        done
+      env:
+        VERSIONS: ${{ inputs.ckan-compatible-versions }}
+
+    - name: Setup CKAN Filter
+      shell: bash
+      if: inputs.ckan-filters != ''
+      run: |
+        for filter in $FILTERS; do
+        ckan filter add "$filter"
+        done
+      env:
+        FILTERS: ${{ inputs.ckan-filters }}
+
+    - name: Install CKAN Mods
```

Comment:
Is there a reason this is here and not in install-dependencies?

Reply:
I was thinking install-dependencies is just for the msbuild target. maybe it should be named better

Reply:
oooh or, I could get rid of that entirely. have the install dependencies run after `restore` and then just `dotnet restore` will call ckan to install everything

Reply:
That's a good idea.  I think there's still a use case for an action that can install mods from ckan (possibly especially for non-code mods) but I could see most code-based mods using that approach.

Reply:
Yeah agreed. You think it should be standalone?